src/wfs/src/main/java/org/geoserver/wfs/kvp/FilterKvpParser.java

```diff
@@ -81,7 +82,9 @@ public Object parse(String value) throws Exception {
                 InputStream input = new ByteArrayInputStream(string.getBytes());
 
                 try {
-                    Filter filter = (Filter) parser.parse(input);
+                    InputSource inputSource = new InputSource(input);
+                    inputSource.setEncoding(Charset.defaultCharset().name());
```

Comment:
Hi, I have similar problems when the GetFeature request wants to autodetect the charset of the input text. The UTF-8 charset is used by default when this autodetection fails.

See:
https://github.com/geoserver/geoserver/blob/master/src/ows/src/main/java/org/geoserver/ows/util/XmlCharsetDetector.java#L330

I must to append the first line with the encoding.

```
<?xml version="1.0" encoding="ISO-8859-1" ?>

<wfs:GetFeature service="WFS" version="1.1.0"
  xmlns:topp="http://www.openplans.org/topp"
  xmlns:wfs="http://www.opengis.net/wfs"
  xmlns:ogc="http://www.opengis.net/ogc"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.1.0/wfs.xsd">
  
  <wfs:Query typeName="IDENA:CATAST_Pol_Municipio">
    
    <ogc:Filter xmlns:ogc="http://www.opengis.net/ogc">
          <PropertyIsLike escape="\" singleChar="_" wildCard="*" matchCase="false">
            <PropertyName>MUNINOAC</PropertyName>
            <Literal>*BARAÑA*</Literal>
          </PropertyIsLike>
    </ogc:Filter>    
  </wfs:Query>
</wfs:GetFeature>
```

It can also be a good place to consider this as well?